sdks/python/apache_beam/io/textio.py

```diff
@@ -675,7 +686,8 @@ def __init__(
         escapechar=escapechar)
 
   def expand(self, pvalue):
-    return pvalue.pipeline | Read(self._source)
+    return pvalue.pipeline | Read(self._source).with_output_types(
+        self._source.output_type_hint())
```

Comment:
Just a note: this is a non-backward-compatible change and will break if the user code sets `with_output_types` especially when it's an auto-generated Row schema (for Beam >= v2.39.0, the fix on user-side is: users no longer need to set `with_output_types`)

Reply:
We should document this somewhere explicitly.